plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/HeteroDescribableConfigurator.java

```diff
@@ -118,18 +120,23 @@ public CNode describe(T instance, ConfigurationContext context) {
     }
 
     @CheckForNull
-    public String describeStructure(T instance, ConfigurationContext context) {
+    public Map<String,List<String>> describeStructure(T instance, ConfigurationContext context) {
+        Map<String, List<String>> heteroConfigurationMapper = new LinkedHashMap<>();
         lookupConfigurator(context, instance.getClass())
             .map(configurator -> convertToNode(context, configurator, instance))
             .filter(Objects::nonNull)
             .map(node -> {
-                if(node.getClass().isEnum()){
-                    return instance.getDescriptor().getDisplayName();
-            } else {
-                     return null;
-                }
+
+                //Could something like this be done
+                    String BaseConfiguratorNode =  node.getClass().getName();
+                    List<String> propertiesList = new ArrayList<>();
+                    for(CNode cNode: node) {
+                        propertiesList.add(cNode.toString());
+                    }
```

Comment:
@casz This is the concept in my mind where I could have a map for the base configurator for every heteroDescribable Configurator and all of its properties nested as values in a map.
The problem I havent put much thought into is how would we deal with further nesting considering that each and every property at max has more levels of nesting.

Reply:
the nesting is handled by calling `lookupConfigurator` and `describeStructure` until there is no more in that structure to be described.

We are basically using the Configurator interface to provide us with the necessary nesting much like how the current implementation is for describing the databinding with `describe`

We can properly reuse `CNode` to store the structure and once we have configured all CNodes we turn the `CNode` into JSON much like we do for describing databinding into YAML.

However the CNode will be carrying information about `property` and `type` instead of `property` and `value`.